src/coprocessor/codec/datum.rs

```diff
@@ -82,12 +83,16 @@ impl Debug for Datum {
     }
 }
 
+/// `cmp_f64` compares the f64 values and returns the Ordering.
 #[inline]
 pub fn cmp_f64(l: f64, r: f64) -> Result<Ordering> {
     l.partial_cmp(&r)
         .ok_or_else(|| invalid_type!("{} and {} can't be compared", l, r))
 }
 
+/// `checked_add_i64` checks whether `r` is larger or equal to 0.
+/// If yes, it adds `r` to the `l`, otherwise, it gets the negated version
+/// of `r` and reduces it from `l`.
```

Comment:
Could we make it easier to understand like:
 `checked_add_i64` checks and adds `r` to the `l`. Return None if the sum is negative.